pipelines/zendesk/__init__.py

```
@@ -0,0 +1,221 @@
+        resource_api = resource_api_method()[::per_page]
+    except TypeError:
+        # this resource doesn't support slicing for pagination
+        resource_api_method = getattr(zendesk_client, f"{resource}")
+        resource_api = resource_api_method()
+    except Exception as e:
+        logger.warning(f"Skipped resource: {resource}. Couldn't load!")
+        logger.warning(str(e))
+        return
+    dict_generator = basic_load(resource_api=resource_api)
+    yield from dict_generator
+
+
+def incremental_load_resource(zendesk_client: Zenpy, resource: str, per_page: int = 1000, start_date: DateTime = FIRST_DAY_OF_CURRENT_YEAR) -> Iterator[TDataItem]:
+    """
+    Basic loader for all Zenpy endpoints that support incremental loads. Supports pagination. Expects to be called as a DLT Resource
+    @:param zendesk_client: Zenpy type object, used to make calls to Zendesk API through zenpy module
+    @:param resource: String that references the zenpy endpoint, usually directly linked to a ZendeskAPI endpoint, ex: users
+    @:param per_page: Indicated how many resources should be retrieved per page with the default being 1000
+    @:param start_date: Datetime object that serves as the starting date of when to retrieve objects
+    @:returns: Generator of dicts
+    """
+
+    try:
+        # use the zenpy endpoint with methods to yield the data
+        resource_api_method = getattr(zendesk_client, f"{resource}")
+        resource_api_method_incremental = resource_api_method.incremental
+        resource_api = resource_api_method_incremental(start_date=start_date, per_page=per_page)
+    except TypeError:
+        # this resource doesn't support incremental so just load normally
+        resource_api_method = getattr(zendesk_client, f"{resource}")
+        resource_api = resource_api_method()
+    except Exception as e:
+        logger.warning(f"Skipped resource: {resource}. Couldn't load!")
```

swallowing exceptions looks really dangerous. you should log and then raise